lib/backend/k8s/resources/watcher.go

```diff
@@ -128,14 +121,9 @@ func (crw *k8sWatcherConverter) processK8sEvents() {
 				case crw.resultChan <- *e:
 					crw.logCxt.Debug("Kubernetes event converted and sent to backend watcher")
 
-					// If this is an error event, check to see if it's a terminating one (the
-					// convertEvent method will decide that).  If so, terminate this watcher.
+					// If this is an error event, bubble up the watcher error
```

Comment:
i think this comment is not necessary - all you are doing in the the branch below is logging.  Perhaps just remove this branch completely - it doesn't feel necessary to log here since the calling code can log the event details if it needs to.